packages/bigquery/src/dataset.js

```diff
@@ -178,6 +178,17 @@ function Dataset(bigQuery, id) {
   });
 
   this.bigQuery = bigQuery;
+
+  this.interceptors.push({
+    request: function(reqOpts) {
+      if (reqOpts.method === 'PATCH' && reqOpts.json.etag) {
+        reqOpts.headers = reqOpts.headers || {};
+        reqOpts.headers['If-Match'] = reqOpts.json.etag;
```

Comment:
What methods is this used for? I think localizing this to the methods themselves would be better, so no magic is happening.

Reply:
The way it is documented in the upstream docs, it applies for any method that makes a `PATCH` request.

https://cloud.google.com/bigquery/docs/api-performance#read-patch-write

Reply:
Ah okay. Can you leave that link in a code comment?